### PR TITLE
Android/search page filter

### LINKED ISSUE #576: [Android] Implementing Search Filter
* **Project:ANDROID**

* **This is a:FEATURE REQUEST**

* **Description of the issue**

Adding search filter to search page.

* **Deadline for resolution:** 

26.01.2020

### PR DESCRIPTION
**Pull request by rambo250**
fixes #576

### CODE REVIEW

File: android/app/src/main/java/com/bounswe2020group3/paperlayer/search/AdvancedSearchFragment.kt [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    }
+
+    private fun searchSubmit(keyword:String){
+        writeLogMessage("i", TAG,"Advanced Search : $keyword submitted .")
+        onSearchButtonClicked(keyword)
+    }
+
+    private fun onSearchButtonClicked(keyword:String){
+        writeLogMessage("i", TAG,"Advanced Search : $keyword submitted .")
+        //val tagIdList:ArrayList<Int> = ArrayList<Int>()
```
Comment (yusufbayam) [inline android/app/src/main/java/com/bounswe2020group3/paperlayer/search/AdvancedSearchFragment.kt:168]: Can you remove comment lines please?

File: android/app/src/main/java/com/bounswe2020group3/paperlayer/search/AdvancedSearchFragment.kt [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+        val checkedChipIds = fragmentView.chipGroupTags.checkedChipIds
+       // for(id in checkedChipIds){
```
Comment (yusufbayam) [inline android/app/src/main/java/com/bounswe2020group3/paperlayer/search/AdvancedSearchFragment.kt:170]: Can you remove comment lines please?

File: android/app/src/main/java/com/bounswe2020group3/paperlayer/search/AdvancedSearchFragment.kt [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+                    projectDueDateBefore=null
+                if(projectEvent=="")
+                    projectEvent=null
+                if(projectState=="")
+                    projectState= null
+                if (projectTags != null) {
+                    if(projectTags.isEmpty())
+                        projectTags=null
+
+                    //writeLogMessage("i", TAG,"Advanced Search : $projectState + $projectEvent + $projectDueDateAfter + $projectDueDateBefore submitted .")
```
Comment (yusufbayam) [inline android/app/src/main/java/com/bounswe2020group3/paperlayer/search/AdvancedSearchFragment.kt:195]: Can you remove comment lines please?

File: android/app/src/main/java/com/bounswe2020group3/paperlayer/search/AdvancedSearchFragment.kt [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+        //}
+        writeLogMessage("i",TAG,"Chips "+ checkedChipIds.size)
+        when(this.tabLayoutSearchFilter.selectedTabPosition){
+            0->{//Project Tab selected
+                var projectDueDateAfter: String? = this.fragmentView.editTextProjectDueDateAfter.editText?.text.toString()
+                var projectDueDateBefore: String? = this.fragmentView.editTextProjectDueDateBefore.editText?.text.toString()
+                var projectEvent:String?= this.fragmentView.editTextProjectEvent.editText?.text.toString()
+                var projectState:String? =this.fragmentView.autoCompleteTextViewProjectState.text.toString()
+                var projectTags:List<Int>? = this.fragmentView.chipGroupTags.checkedChipIds
+                // writeLogMessage("i",TAG,"Project Dua Date Before "+projectDueDateBefore)
```
Comment (yusufbayam) [inline android/app/src/main/java/com/bounswe2020group3/paperlayer/search/AdvancedSearchFragment.kt:182]: Can you remove comment lines please?

File: android/app/src/main/java/com/bounswe2020group3/paperlayer/search/AdvancedSearchFragment.kt [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+
+    private fun initTagsGroup(tagList: List<Tag>){
+
+        //Adding tags to dynamically to project tags
+        if (this.fragmentView.chipGroupTags?.childCount != tagList.size) {
+            for (tag in tagList){
+                val chip = Chip(this.fragmentView.chipGroupTags.context)
+                chip.text=tag.name
+                chip.id=tag.id
+                //chipIdToTagId[chip.id]=tag.id
```
Comment (yusufbayam) [inline android/app/src/main/java/com/bounswe2020group3/paperlayer/search/AdvancedSearchFragment.kt:256]: Can you remove comment lines please?

**Review by yusufbayam — COMMENTED** [review]
Looking good in general, we can merge after resolving issues?

File: android/app/src/main/java/com/bounswe2020group3/paperlayer/util/Constants.kt [hunk 6]
```diff
@@ -1,5 +1,5 @@
 package com.bounswe2020group3.paperlayer.util
 
-//const val BASE_URL = "https://paperlayer.herokuapp.com"
-const val BASE_URL = "https://paperlayer-backend.azurewebsites.net"
-const val TOKEN = "Token 94636236e607c560cb13333e4b60b9faa5bc125d"
\ No newline at end of file
+const val BASE_URL = "https://paperlayer.herokuapp.com"
+//const val BASE_URL = "https://paperlayer-backend.azurewebsites.net"
```
Comment (ybedirhanpak) [inline android/app/src/main/java/com/bounswe2020group3/paperlayer/util/Constants.kt:4]: Just like I commented on @mhrfky 's pull requests, I think we should keep the azure in the master

**Review by ybedirhanpak — APPROVED** [review]

**Review by yusufbayam — APPROVED** [review]

**Pull request merged**